src/pdn/src/PdnGen.tcl

```diff
@@ -6253,6 +6281,38 @@ proc generate_voltage_domain_rings {core_ring_data} {
           add_stripe $layer "GROUND_$ground_net" $lower_outer_ring
           add_stripe $layer "GROUND_$ground_net" $upper_outer_ring
         }
+        
+        #create power rings for other power if exists
+        foreach power_temp $power_net {
```

Comment:
Cant we do this as a loop over all nets? When its two rings being added, using a loop is a bit over the top, but now it's 3, potentially more later, I think it makes sense for this to be done in a loop

Reply:
The reason I use two loops is that I want to make sure the inner two power rings are primary_power and primary_ground of the voltage domain. Now I define `-primary_power "VDD VREG"` in pdn.cfg. Maybe I should change it into `-primary_power VDD` `-additional_power VREG`.